File: frontend/src/microservices/flight/objects/GetFlightDetailsPlannerRequest.ts
// 本文件定义 flight 模块的 `GetFlightDetailsPlannerRequest`，作为 planner 请求参数并提供 JSON 编解码。

export type GetFlightDetailsPlannerRequest = {
  flightId: string
}

export const getFlightDetailsPlannerRequestFromJson = (json: string): GetFlightDetailsPlannerRequest =>
  JSON.parse(json) as GetFlightDetailsPlannerRequest

export const getFlightDetailsPlannerRequestToJson = (value: GetFlightDetailsPlannerRequest): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/flight/objects/GetFlightDetailsPlannerRequest.scala
// GetFlightDetailsPlannerRequest defines the flight details planner request model.
package com.typesafe.travel.flight.objects

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class GetFlightDetailsPlannerRequest(flightId: String)

object GetFlightDetailsPlannerRequest:
  given sourceEncoder: Encoder[GetFlightDetailsPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[GetFlightDetailsPlannerRequest] = deriveDecoder

